client.go

```diff
@@ -70,26 +78,40 @@ type Client struct {
 }
 
 // NewClient returns a new API Client using the given APIKey
-func NewClient(apiKey string) *Client {
+func NewClient(apiKey string, options ...ClientOptions) *Client {
+	host := APIHost
+	for _, option := range options {
+		if option.region == "eu" {
```

Comment:
This code is a little problematic because if someone specifies an invalid region, the value is essentially ignored. In go, it's best to err out in that case. Using the strongly typed constants, we don't have to worry about that as much. If we start to have options that can potentially be invalid, we need to also return an error as part of our method signature for `NewClient`.

Also, the array of ClientOptions would allow the region to be specified an unlimited number of times, which is a little awkward.

Reply:
with your change suggestion this code is no longer necessarily.